### PR TITLE
PR 1534 LLVM code coverage

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -67,14 +67,21 @@ jobs:
...
-      - name: Run Tarpaulin
-        uses: actions-rs/tarpaulin@v0.1
-        with:
-          args: '--locked --jobs 16 --timeout 600 --skip-clean -- --test-threads 16'
+      - name: Run cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Generate code coverage
+        run: cargo llvm-cov --workspace --lcov --output-path lcov.info
+        # with:
+        #   args: '--locked --jobs 16 --timeout 600 --skip-clean -- --test-threads 16'
```
Comment (lemmih) [inline .github/workflows/rust.yml:79]: `cargo llvm-cov` takes `--jobs` parameter. We might as well set it to 16.

Comment (lemmih) [reply .github/workflows/rust.yml:79]: Oh wait, it defaults to the number of available cpus. Yeah, let's leave it then.

Comment (lemmih) [reply .github/workflows/rust.yml:79]: Let's just remove the commented-out old arguments since we're not using them anymore.

**Comment by LesnyRumcajs** [comment]
Any high-level wrap-up on the differences and how is it better than `tarpaulin`? Seems the coverage will drop, is this more correct than the current solution?

**Comment by lemmih** [comment]
> Any high-level wrap-up on the differences and how is it better than `tarpaulin`? Seems the coverage will drop, is this more correct than the current solution?

llvm-cov instruments the source code and will always generate 100% accurate results, even if optimizations are enabled. Tarpaulin interrupts the program every so often to look at the call-stack. It is not 100% accurate at the best of times and optimizations ruin the results entirely. The reason we didn't use it earlier is because it just became stable in rust-1.60.

**Comment by LesnyRumcajs** [comment]
@lemmih sounds good, seems the coverage dropped because more lines are now taken into consideration.

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]
Looks good, but this file needs to be removed: https://github.com/ChainSafe/forest/blob/main/.tarpaulin.toml

**Comment by lemmih** [comment]
> Looks good, but this file needs to be removed: https://github.com/ChainSafe/forest/blob/main/.tarpaulin.toml

Ah, right. @tyshko5 could you see to that?

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**